cuML/src/glm/qn/glm_regularizer.h

```diff
@@ -58,12 +59,19 @@ template <typename T, class Loss, class Reg> struct RegularizedGLM : GLMDims {
                         const SimpleMat<T> &Xb, const SimpleVec<T> &yb,
                         SimpleMat<T> &Zb, cudaStream_t stream, bool initGradZero = true) {
     SimpleVec<T> lossVal(loss_val, 1);
-    G.fill(0);
+    G.fill(0, stream);
     reg->reg_grad(lossVal.data, G, W, loss->fit_intercept, stream);
-    T reg = lossVal[0];
+
+    T reg_host; 
+    MLCommon::updateHostAsync(&reg_host, lossVal.data, 1, stream);
+    CUDA_CHECK(cudaStreamSynchronize(stream));
```

Comment:
`reg_host` is not used before line 74 so one `cudaStreamSynchronize` in line 72 is sufficient and the `cudaStreamSynchronize` in line 67 is unnecessary.